### PR TITLE
Add Python server

### PR DESCRIPTION
**Pull request by tamsingreen**
**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      
https://github.com/bbc/digital-paper-edit-infrastructure/issues/21 

**Describe what the PR does**    
- Add a simple Python server for use on the deployed DPE EC2. Works in conjunction with this [PR](https://github.com/bbc/digital-paper-edit-infrastructure/pull/22) in the infrastructure project. It has a /status healthcheck endpoint for AWS use, and all other requests fall back to /index.html where the React app is served from.
- Adds an isProduction() function to check whether Dev/Test or Prod/Live config should be used (this is added through the build process in the digital-paper-edit-infrastructure project).
- Changes some `const` variables to `let`where reassignment was causing read-only errors when deployed

**State whether the PR is ready for review or whether it needs extra work**    
RFR

**How to test**
1. `npm run build`
2. `cd build`
3. `python server.py`
4. Go to http://localhost:8080
5. Check the app name appears in the page header (this shows that env variables are working correctly - ask Tamsin or Alli for the updated config)
6. Log in and create a new project, update the name of a project, and add a new programme script to a project (this shows the read-only errors are fixed)

### CODE REVIEW

File: public/server.py [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
+import SimpleHTTPServer, SocketServer
+class MyHttpRequestHandler(SimpleHTTPServer.SimpleHTTPRequestHandler):
     def do_GET(self):
         if self.path == '/status':
             self.send_response(200)
-        else: 
-            self.path = '/index.html'
-            return http.server.SimpleHTTPRequestHandler.do_GET(self)
+        else:
+            return SimpleHTTPServer.SimpleHTTPRequestHandler.do_GET(self)
```
Comment (emettely) [inline public/server.py:8]: So here, it's calling the inherited function - I think you can do something like `super().do_GET()` 🤔 
Which means you're calling the parent function, and I think the "self" call might be implicit.

Obviously if the code works I'm also quite happy as it is, just curious I suppose :)

**Review by emettely — COMMENTED** [review]
Is this already deployed?

**Comment by tamsingreen** [comment]
> Is this already deployed?

No, the infra project is pulling from this branch.

File: public/server.py (hunk 1 above)
Comment (tamsingreen) [reply public/server.py:8]: This is a very cut and pasted solution so any improvements welcome, thanks!

**Comment by emettely** [comment]
> > Is this already deployed?
> 
> No, the infra project is pulling from this branch.

This probably should be put in the infra repository than this one. I say this because this has not very much to do with the actual Firebase / open source project, but a lot to do with the AWS / Cosmos side of things.

File: public/server.py (hunk 1 above)
Comment (tamsingreen) [reply public/server.py:8]: I tried this suggestion and it didn't work :(

**Comment by tamsingreen** [comment]
> > > Is this already deployed?
> > 
> > 
> > No, the infra project is pulling from this branch.
> 
> This probably should be put in the infra repository than this one. I say this because this has not very much to do with the actual Firebase / open source project, but a lot to do with the AWS / Cosmos side of things.

This is true but it's a simple, generic server with no BBC-specific config so I'm happy for it to belong in this repo.

**Review by allishultes — APPROVED** [review]
This looks good to me, Tamsin - hero status for your fixes with the lets / consts!

File: src/Util/is-production/index.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import url from 'url';
+
+const isProduction = () => {
+  const parsedUrl = url.parse(window.location.href, true);
+
+  if ((parsedUrl.host) === 'digital-paper-edit.tools.bbc.co.uk') {
```
Comment (sarahrainbow) [inline src/Util/is-production/index.js:6]: Could this be simplified? e.g. 
``` 
return parsedUrl.host === 'digital-paper-edit.tools.bbc.co.uk'
```

or does this make it less clear?

**Review by sarahrainbow — APPROVED** [review]
Looks great and works for me! Just had one comment that may not need addressing!

File: public/server.py [hunk 3]
```diff
@@ -1,17 +1,15 @@
 #!/usr/bin/env python
-import http.server
-import socketserver
-class MyHttpRequestHandler(http.server.SimpleHTTPRequestHandler):
+import SimpleHTTPServer, SocketServer
```
Comment (aniablaziak) [inline public/server.py:2]: Does this need to be run using python2? Why is that?

Using python3 I had to change it back to:
`import http.server, socketserver`

Comment (tamsingreen) [reply public/server.py:2]: Good q - the EC2 we use has python 2 by default (as does my Mac :) )

**Comment by bevand10** [comment]
Don’t understand the need for an external dependency on url when window.location.host already provides that value?

--Dave


Dave Bevan
Senior Systems Engineer, News Labs & Tech Lead, Internet Fit Radio

BBC News Labs

+44 (0) 7703 54 66 92
New Broadcasting House | 2nd floor | Zone B | London W1A 1AA

@BBC_News_Labs<https://twitter.com/bbc_news_labs?lang=en>
bbcnewslabs.co.uk<https://bbcnewslabs.co.uk/>

On 28 Apr 2021, at 15:55, Sarah Rainbow ***@***.***> wrote:

﻿

@sarahrainbow commented on this pull request.

________________________________

In src/Util/is-production/index.js<https://github.com/bbc/digital-paper-edit-firebase/pull/209#discussion_r622262175>:

> @@ -0,0 +1,14 @@
+import url from 'url';
+
+const isProduction = () => {
+  const parsedUrl = url.parse(window.location.href, true);
+
+  if ((parsedUrl.host) === 'digital-paper-edit.tools.bbc.co.uk') {


Could this be simplified? e.g.

return parsedUrl.host === 'digital-paper-edit.tools.bbc.co.uk'


or does this make it less clear?

**Pull request merged**